src/components/TableComponent/index.jsx

```diff
@@ -1,166 +1,120 @@
+        ordering,
       });
+      sortTable(ordering);
       sendEnterpriseTrackEvent(enterpriseId, 'edx.ui.enterprise.admin_portal.table.sorted', {
         tableId: id,
-        column: column.label,
-        direction,
+        column: column.id,
+        direction: column.desc ? 'desc' : 'asc',
       });
-    };
-
-    const columnConfig = this.props.columns.map(column => ({
-      ...column,
-      onSort: column.columnSortable ? (direction) => sortByColumn(column, direction) : null,
-    }));
-
-    let sortDirection;
-    let sortColumn;
-
-    if (tableSortable) {
-      sortDirection = ordering && ordering.indexOf('-') !== -1 ? 'desc' : 'asc';
-      sortColumn = (ordering && ordering.replace('-', '')) || columnConfig[0].key;
     }
+  }, []);
 
-    return (
-      <div className={className}>
-        <div className="row">
-          <div className="col">
-            {loading && <TableLoadingOverlay />}
-            <div className="table-responsive">
-              <Table
-                id={id}
-                className="table-sm table-striped"
-                columns={columnConfig}
-                data={formatData(data)}
-                tableSortable={tableSortable}
-                defaultSortedColumn={sortColumn}
-                defaultSortDirection={sortDirection}
-              />
-            </div>
-          </div>
-        </div>
-        <div className="row mt-2">
-          <div className="col d-flex justify-content-center">
-            <Pagination
-              paginationLabel={`${id}-pagination`}
-              pageCount={pageCount}
-              currentPage={currentPage}
-              onPageSelect={(page) => {
-                updateUrl({ page });
-                sendEnterpriseTrackEvent(enterpriseId, 'edx.ui.enterprise.admin_portal.table.paginated', {
-                  tableId: id,
-                  page,
-                });
-              }}
-            />
+  useEffect(() => {
+    // Get initial data
+    paginateTable();
+    return clearTable;
+  }, []);
+
+  const handlePageChange = (page) => {
+    updateUrl({ page });
+    sendEnterpriseTrackEvent(enterpriseId, 'edx.ui.enterprise.admin_portal.table.paginated', {
+      tableId: id,
+      page,
+    });
+    paginateTable(parseInt(page, 10));
+  };
+
+  const renderTableContent = () => (
+    <div className={className}>
+      <div className="row">
+        <div className="col">
+          <div className="table-responsive">
+            <DataTable
+              id={id}
+              className="table-sm table-striped"
+              columns={columns}
+              data={formatData(data)}
+              isSortable={tableSortable}
+              itemCount={totalCount}
+              isLoading={loading}
+              manualSortBy
+              fetchData={sortByColumn}
+            >
+              <DataTable.TableControlBar />
+              <DataTable.Table />
+              <DataTable.TableFooter>
```

Comment:
I believe we could rely on the stock `<DataTable.TableFooter />` here rather than defining our own. If we do need to define our own, we should be using the `DataTable`-specific pagination component as otherwise the pagination isn't properly wired up with how `DataTable` is supposed to be used.